src/freenet/pluginmanager/PluginManager.java

```diff
@@ -409,17 +395,17 @@ private PluginInfoWrapper realStartPlugin(final PluginDownLoader<?> pdl, final S
 		FredPlugin plug;
 		PluginInfoWrapper pi = null;
 		try {
-			plug = loadPlugin(pdl, filename, pluginProgress, ignoreOld);
-			if (plug == null)
-				return null; // Already loaded
-			pluginProgress.setProgress(PluginProgress.PROGRESS_STATE.STARTING);
+			plug = loadPlugin(pdl, filename, pluginProgress, alwaysDownload);
+			pluginProgress.setProgress(ProgressState.STARTING);
 			pi = new PluginInfoWrapper(node, plug, filename, pdl.isOfficialPluginLoader());
 			PluginHandler.startPlugin(PluginManager.this, pi);
+			pluginWrappers.add(pi);
 			synchronized (pluginWrappers) {
-				pluginWrappers.add(pi);
```

Comment:
This should remain in the `synchronized` block, for `pluginWrappers` and `pluginsFailedLoad` to remain consistent.